### DISCUSSION TITLE
Upload of 150MB file to a Release fails - max 25MB allowed?

### DISCUSSION THREAD

**Discussion by orb15** [community/community#146417] [Repositories] [answered]
### Select Topic Area

Question

### Body

I am working on a project where, in addition to sharing my code via a Release, I need to include a ZIP file of content associated with that release. Note that the content is not part of the repository data per se; I am not talking about uploading large files to be tracked as part of the repo, only on uploading large files as part of a release.

My reading of the release documentation indicated that no one release file can exceed 2GiB but when I attempt to upload my 150MB zip file in the release creation web page, GitHub refuses to accept the upload and indicates that uploaded files cannot exceed 25MB. 

Am I doing something wrong? Do I need GLFS? A paid account?

I am currently working through a browser on the free tier if that matters.

(edited for clarity after initial submission)

**Comment by hrittm** [comment, marked as answer]
It sounds like you're running into trouble uploading your 150 MB file to a GitHub release. Here's the deal:

###  1. File Size Limit
GitHub lets you **upload files up to 2 GB for releases, so a 150 MB file should work fine.** The "25 MB" limit you saw might be for other uploads, like in comments or issues.

###  2. Why It’s Not Working
If it’s failing, it could be a browser or network issue, or even something weird with your account/session.

###  3. How to Fix It
- Switch Browsers: Try a different browser or clear your cache and try again.
- Use GitHub CLI: You can use a tool called GitHub CLI to upload the file instead of the browser. Here’s the command:
```bash
gh release upload <your-tag> <your-file>
```
Example:
```bash
gh release upload v1.0 my-file.zip
```
Check Your Network: Make sure your internet connection is strong.

###  4. If Nothing Works
You can contact GitHub Support or temporarily host your file on another service (like Google Drive) and share the link in the release notes.

Hope this helps!

**Reply by orb15** [reply]
I had tried several times before posting this message, on two different days, but ran into the same issue. Today, after receiving your reply that confirmed I had been doing the process correctly and my reading of the documentation was accurate, I thought I would give it a final try before using the command line approach.  I was fully prepared to do screen shots to show the error I expected to get.

It worked on the first try and may tries thereafter as I polished my release.

I am going to put this down to some kind of networking issue.  Thank you very much for taking the time to answer my question!

**Reply by hrittm** [reply]
> I had tried several times before posting this message, on two different days, but ran into the same issue. Today, after receiving your reply that confirmed I had been doing the process correctly and my reading of the documentation was accurate, I thought I would give it a final try before using the command line approach.  I was fully prepared to do screen shots to show the error I expected to get.
> 
> It worked on the first try and may tries thereafter as I polished my release.
> 
> I am going to put this down to some kind of networking issue.  Thank you very much for taking the time to answer my question!
> 
> 

## Glad to hear it worked out for you! 
Sometimes these things really do come down to temporary glitches or network issues, as frustrating as that can be. Thanks for letting me know—it’s great that your release is all set now!

If you run into anything else in the future, don’t hesitate to ask. Good luck with your project!

**Reply by gajop** [reply]
This definitely happened to me as well, and much like you, it only fixed itself after reloading the page - the file I tried to upload was the same.

**Reply by tribes2** [reply]
Yummy AI slop!

**Comment by Abhishekkmsharmawebdev** [comment]
🔹 Step-by-Step Instructions:
Install GitHub CLI (if not installed):
https://cli.github.com/

Login via CLI:

Code:
gh auth login
Create a Release (if not already created):


gh release create v1.0.0 path/to/your-bigfile.zip --title "My Release" --notes "Release notes here"
Replace v1.0.0 with your actual tag. If it doesn’t exist, this will create it.

Or, Upload to Existing Release:


gh release upload v1.0.0 path/to/your-bigfile.zip

**Comment by NTHoang2103** [comment]
Since your 150MB file exceeds the 25MB web limit, you must use the GitHub API to perform the upload. The most common and easiest method is by using the command-line tool GitHub CLI (gh).

1. Preparation (Installation & Login)
Install GitHub CLI: Download and install the gh tool from the GitHub CLI homepage (search for "GitHub CLI install").

Log in to GitHub CLI: Open your Terminal (or Command Prompt) and run the following command to authenticate your GitHub account:

Bash

gh auth login
Follow the on-screen instructions to complete the authentication process.

2. Upload to an Existing Release
Assuming you have already created a Draft Release on GitHub.

Open your Terminal and navigate to the directory containing your 150MB file.

Run the following command:

Bash

gh release upload <tag-name> <path-to-your-file>
Explanation:

Replace <tag-name> with the tag name of the Release you want to attach the file to (e.g., v1.0.0).

Replace <path-to-your-file> with the actual name of your 150MB file (e.g., App_Setup_150MB.zip).

Example:

Bash

gh release upload v1.0.0 App_Setup_150MB.zip
3. Creating and Uploading Simultaneously (Optional)
If you haven't created the Release yet, you can create it and attach the file in a single command:

Bash

gh release create v1.1.0 --repo <owner/repo> -t "Final Release 1.1.0" -n "Added new features." App_Setup_150MB.zip
By using GitHub CLI, you can easily upload files up to 2 GB per asset in your Release.
